### ISSUE TITLE
PLE1307: false positives

### ISSUE THREAD

**Issue by spaceone** [charliermarsh/ruff#2724]
```sh
$ echo "import random; '%d' % (random.randint(0, 100),)" | ruff --isolated --select PLE1307 -
-:1:16: PLE1307 Format type does not match argument type
```

further ones:
`"-%f" % time.time()`
`"%r" % (object['dn'],)`
`r'\%03o' % (ord(c),)`
`('%02X' % int(_) for _ in o)`
`"%s;range=%d-*" % (attr, upper + 1)`
`"%d" % (len(foo),)`

**Comment by charliermarsh** [comment]
Cool, we'll either fix these or disable the rule for now prior to the next release.
